### PR TITLE
Changelog for 4.0.0-alpha.1

### PR DESCRIPTION
**Pull request by BoD**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,136 @@
...
+    
+    // Get the generated schema types (and fragments) from the upstream schema module 
+    dependsOn(project(":schema")) 
+  }
+}
+```
+
+## ⚙️ Other changes
+
+- All the previously deprecated APIs have been removed.
```
Comment (martinbonnin) [inline CHANGELOG.md:130]: I don't think that is technically true? We should allow symbols to stay deprecated for some time before removing them so symbols that were deprecated in the late 3.x should still be there?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,136 @@
...
+  .enqueue(new ApolloCallback<MyQuery.Data>() {
+      @Override public void onResponse(@NotNull ApolloResponse<MyQuery.Data> response) {
+        System.out.prinitln(response.getData());
+      }
+  });
+```
+
+A few examples can be found in the [tests](https://github.com/apollographql/apollo-kotlin/tree/main/tests/java-client/src/test/java/test).
+
+## 🔃 Multi-module improvements
```
Comment (martinbonnin) [inline CHANGELOG.md:88]: Add this section to the migration guide as well?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,136 @@
...
+    // Get the generated schema types (and fragments) from the upstream schema module 
+    dependsOn(project(":schema")) 
+  }
+}
+```
+
+## ⚙️ Other changes
+
+- All the previously deprecated APIs have been removed.
+- For a more exhaustive list of changes, see the [migration guide](https://www.apollographql.com/docs/kotlin/v4/migration/4.0).
```
Comment (martinbonnin) [inline CHANGELOG.md:131]: That's a bit confusing as the migration guide doesn't really contain all the changes (doesn't contain new features for an example). All in all, I think I would remove the "other changes" section 


About migration guide, might be worth doing a quick path on it? Add the multi-module stuff + it contains some details about HTTP headers that I'm not 100% sure are still valid

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,136 @@
...
 
+# Version 4.0.0-alpha.1
+
+_2023-05-15_
+
+This release is the first alpha of the next major version of Apollo Kotlin: 4.0.0. 
+
+This version is under development, but we want to give you a preview of what's coming, and get your early feedback. Please see the [roadmap](https://github.com/apollographql/apollo-kotlin/blob/main/ROADMAP.md) for more details about the release plan.
+
+As this is a major version, this release introduces new features but also some breaking changes.
```
[the remark this answers is not in the document]
Comment (BoD) [reply CHANGELOG.md:12]: 👍 Good call about the 3->4 diff being smaller than 2->3!

File: CHANGELOG.md (hunk 1 above)
Comment (BoD) [reply CHANGELOG.md:130]: Yeah I may misremember something - let me refresh my memory here.

**Review by martinbonnin — APPROVED** [review]

**Pull request merged**